lib/SPIRV/libSPIRV/SPIRVBasicBlock.cpp

```diff
@@ -69,6 +69,15 @@ SPIRVBasicBlock::addInstruction(SPIRVInstruction *I) {
   return I;
 }
 
+SPIRVVariable *
+SPIRVBasicBlock::addVariable(SPIRVVariable *Var) {
+  assert(Var && "Invalid variable");
+  Module->add(Var);
+  Var->setParent(this);
+  InstVec.insert(InstVec.begin(), Var);

```

Comment:
This will cause auto vars in source code emitted in reverse order and break existing lit tests.

Should find the first non-Variable instruction and insert before that.